elixir/apps/domain/lib/domain/okta/sync.ex

```diff
@@ -0,0 +1,815 @@
+        raise Okta.SyncError,
+          reason: "Failed to stream group members",
+          cause: reason,
+          directory_id: directory_id,
+          step: :stream_group_members
+    end)
+    |> Enum.reverse()
+  end
+
+  # Helper to build issuer URL
+  defp issuer(directory), do: "https://#{directory.okta_domain}"
+
+  # Parses an Okta user API response into a structured map
+  defp parse_okta_user(user) do
+    profile = user["profile"] || %{}
+
+    first_name = (profile["firstName"] || "") |> String.trim()
+    last_name = (profile["lastName"] || "") |> String.trim()
+    email = profile["email"] |> String.downcase() |> String.trim()
+
+    %{
+      okta_id: user["id"],
+      email: email,
+      first_name: first_name,
+      last_name: last_name,
+      full_name: "#{first_name} #{last_name}"
+    }
+  end
+
+  # Parses an Okta group API response into a structured map
+  defp parse_okta_group(group) do
+    profile = group["profile"] || %{}
+
+    %{
+      okta_id: group["id"],
+      name: profile["name"] || group["id"]
+    }
+  end
+
+  # Delete records that weren't synced this time
+  defp delete_unsynced(directory, synced_at) do
+    account_id = directory.account_id
+    directory_id = directory.id
+
+    # Delete groups that weren't synced
+    {deleted_groups_count, _} =
+      DB.delete_unsynced_groups(account_id, directory_id, synced_at)
+
+    Logger.debug("Deleted unsynced groups",
+      okta_directory_id: directory.id,
+      count: deleted_groups_count
+    )
+
+    # Delete identities that weren't synced
+    {deleted_identities_count, _} =
+      DB.delete_unsynced_identities(account_id, directory_id, synced_at)
+
+    Logger.debug("Deleted unsynced identities",
+      okta_directory_id: directory.id,
+      count: deleted_identities_count
+    )
+
+    # Delete memberships that weren't synced
+    {deleted_memberships_count, _} =
+      DB.delete_unsynced_memberships(account_id, directory_id, synced_at)
+
+    Logger.debug("Deleted unsynced group memberships",
+      okta_directory_id: directory.id,
+      count: deleted_memberships_count
+    )
+
+    # Delete actors that no longer have any identities and were created by this directory
+    {deleted_actors_count, _} = DB.delete_actors_without_identities(account_id, directory_id)
+
+    Logger.debug("Deleted actors without identities",
+      okta_directory_id: directory.id,
+      count: deleted_actors_count
+    )
+  end
+
+  defmodule DB do
+    import Ecto.Query
+    alias Domain.Safe
+
+    def get_directory(id) do
+      from(d in Okta.Directory, as: :directories)
+      |> where([directories: d], d.id == ^id)
+      |> where([directories: d], d.is_disabled == false)
+      |> Safe.unscoped()
+      |> Safe.one()
+      |> case do
+        nil -> {:error, :not_found}
+        directory -> {:ok, directory}
+      end
+    end
+
+    def get_synced_group_idp_ids(account_id, directory_id, synced_at) do
+      from(g in Domain.Actors.Group,
```

Comment:
`Actors.Group` isn't a thing anymore.


```suggestion
      from(g in Domain.Group,
```

Reply:
👍 I'll make sure to get this updated